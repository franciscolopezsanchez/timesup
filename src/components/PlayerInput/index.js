import React, {useState} from "react"

function PlayerInput({teamId, createPlayer}) {
  const [player, setPlayer] = useState("")
  if (isNaN(teamId)) return null
  return (
    <div>
      <input
        type="text"
        value={player}
        onChange={e => setPlayer(e.target.value)}
        onBlur={() => {
          if (player) createPlayer(player, teamId)
          setPlayer("")
        }}
      />
    </div>
  )
}

export default PlayerInput
