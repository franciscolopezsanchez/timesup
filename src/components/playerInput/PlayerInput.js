import React from 'react'

function PlayerInput() {

    return (
        <input type="text"/>
      )
}

export default PlayerInput