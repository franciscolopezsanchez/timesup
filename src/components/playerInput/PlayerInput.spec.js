import React from "react"
import { render, cleanup, fireEvent } from "@testing-library/react"
import PlayerInput from "./PlayerInput"

afterEach(cleanup)

describe("PlayerInput tests", () => {
	test("PlayerInput handles undefined teamId as argument", () => {
		// Arrange
		const teamId = undefined
		// Act
		const { container } = render(<PlayerInput teamId={teamId} />)
		// Assert
		expect(container.firstChild).toBeFalsy()
	})

	test("PlayerInput handles a teamId as argument", () => {
		// Arrange
		const teamId = 0
		// Act
		const { container } = render(<PlayerInput teamId={teamId} />)
		const input = document.querySelector("input")
		// Assert
		expect(container.firstChild).toBeTruthy()
		expect(input).toBeTruthy()
	})

	test("PlayerInput handles an input change", () => {
		// Arrange
		const teamId = 0
		// Act
		const { container } = render(<PlayerInput teamId={teamId} />)
		const input = document.querySelector("input")

		fireEvent.change(input, { target: { value: "Jesus" } })
		// Assert
		expect(input.value).toBe("Jesus")
	})

	test("PlayerInput call callback on blur", () => {
		// Arrange
		const teamId = 0
		const createPlayer = jest.fn()
		// Act
		const { container } = render(
			<PlayerInput teamId={teamId} createPlayer={createPlayer} />
		)
		const input = document.querySelector("input")

		fireEvent.change(input, { target: { value: "Jesus" } })
		// Assert
		expect(input.value).toBe("Jesus")

		fireEvent.blur(input)

		expect(createPlayer.mock.calls.length).toBe(1)
		expect(input.value).toBe("")
	})

	test("renders snapshot", () => {
		// Arrange
		// Act
		const { asFragment } = render(<PlayerInput teamId={0} />)
		//Assert
		expect(asFragment()).toMatchSnapshot()
	})
})
