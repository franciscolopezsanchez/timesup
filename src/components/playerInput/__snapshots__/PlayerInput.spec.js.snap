// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerInput tests renders snapshot 1`] = `
<DocumentFragment>
  <div>
    <input
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
